src/bundle/Resources/public/js/scripts/core/multilevel.popup.menu.js

```diff
@@ -0,0 +1,385 @@
+            }
+
+            return newBranchElement;
+        }
+
+        // eslint-disable-next-line no-unused-vars
+        generateGroupIfNotExists(data, processAfterCreated = () => {}) {
+            const { branchElement, id } = data;
+            const groupElement = branchElement.querySelector(`[data-group-id="${id}"]`);
+
+            if (groupElement) {
+                return groupElement;
+            }
+
+            return this.generateGroup(...arguments);
+        }
+
+        generateGroup(data, processAfterCreated = () => {}) {
+            const { branchElement, id } = data;
+            const { groupTemplate } = this.container.dataset;
+
+            const container = doc.createElement('div');
+            const renderedGroup = groupTemplate.replaceAll('{{ group_id }}', id);
+
+            container.insertAdjacentHTML('beforeend', renderedGroup);
+
+            const newGroupElement = container.querySelector('.ibexa-popup-menu__group');
+
+            processAfterCreated(newGroupElement, data);
+
+            branchElement.appendChild(newGroupElement);
+
+            return newGroupElement;
+        }
+
+        generateItem(data, processAfterCreated = () => {}) {
+            const { label, sublabel = '', branchElement, groupId, href, onClick } = data;
+            const { itemTemplateBtn, itemTemplateLink } = this.container.dataset;
+            const groupElement = this.generateGroupIfNotExists({ branchElement, id: groupId });
+            const itemTemplate = !!href ? itemTemplateLink : itemTemplateBtn;
+
+            const container = doc.createElement('div');
+            const renderedItem = itemTemplate.replaceAll('{{ label }}', label).replaceAll('{{ sublabel }}', sublabel);
+
+            container.insertAdjacentHTML('beforeend', renderedItem);
+
+            const newItemElement = container.querySelector('.ibexa-popup-menu__item');
+            const newItemContentElement = newItemElement.querySelector('.ibexa-popup-menu__item-content');
+
+            if (href) {
+                newItemContentElement.href = href;
+            }
+
+            if (onClick) {
+                newItemContentElement.addEventListener('click', () => onClick(newItemElement, data), false);
+            }
+
+            processAfterCreated(newItemElement, data);
+
+            groupElement.appendChild(newItemElement);
+
+            return newItemElement;
+        }
+
+        getBranchItems(branchElement) {
+            return [...branchElement.querySelectorAll(':scope > .ibexa-popup-menu__group > .ibexa-popup-menu__item')];
+        }
+
+        toggleItemVisibility(menuItem, shouldBeVisible) {
+            const { branchElement } = menuItem;
+
+            menuItem.classList.toggle('ibexa-popup-menu__item--hidden', !shouldBeVisible);
+
+            if (branchElement && !shouldBeVisible) {
+                this.closeWithSubbranches(branchElement);
+            }
+        }
+
+        isOurBranch(branch) {
```

Comment:
what does `Our` mean in this case?

Reply:
Whether this branch belongs to this menu.